Pull Request for Decoding a Morse code message

chuaindia opened the pull request:
In this branch we incur the following:

- A method to decode a Morse code character is created.

- A method to decode an entire word in Morse code is created.

- A method to decode the entire message in Morse code is created.

chuaindia committed:
Add linter files

samuelmoiwa committed:
Update morse_code_decoding.rb

chuaindia committed:
Add ruby file for decoding Morse code

chuaindia committed:
Add decoding file

chuaindia committed:
Modify README file

chuaindia committed:
Put proper indentation

chuaindia committed:
Update the main filefor linters

TedLivist commented on morse_code_decoding.rb:
```diff
@@ -0,0 +1,35 @@
+    @final_word += decode_word(morse_word)
+  end
+
+  puts "#{@final_word} "
```
- [x] Your methods return unnecessary trailing spaces and your `decode` method should return result, not print it. Kindly fix that.

  - You can remove the spacing on line 22 to this: **`"#{word}"`**
  - You can refactor the `decode` method to this to account for the spacing you are looking for:
  
    ```
    def decode(morse_code)
      @final_word = ''
      @morse_code_arr = morse_code.split('   ')
      @morse_code_arr.map do |morse_word|
        @final_word += decode_word(morse_word) + " "
      end
    
      "#{@final_word}".strip
    end
    ```
  
  ![ygf](https://user-images.githubusercontent.com/36542986/208876734-40c94ee5-2172-4bfd-a058-c0be3327c503.jpg)

chuaindia replied:
The necessary changes were made.

TedLivist reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @chuaindia and @samuelmoiwa,

Good job so far!
✔️ Good documentation
✔️ Re-usability of methods
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

chuaindia commented:
I will make the necessary change.


On Wed, 21 Dec 2022, 15:30 Teddy-Livingstone Ememandu, <
***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
> STATUS: CHANGES REQUIRED ♻️ ♻️
>
> Hi @chuaindia <https://github.com/chuaindia> and @samuelmoiwa
> <https://github.com/samuelmoiwa>,
>
> Good job so far!
> ✔️ Good documentation
> ✔️ Re-usability of methods
> However, there are some issues that you still need to work on to go to the
> next project but you are almost there!
> Required Changes ♻️
>
> *Check the comments under the review.*
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear. And please, remember to tag me in your question so I can
> receive the notification.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> *As described in the Code reviews limits policy
> <https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a
> limited number of reviews per project (check the exact number in your
> Dashboard). If you think that the code review was not fair, you can request
> a second opinion using this form <https://airtable.com/shrQAqnBwek5a0O0s>.*
> ------------------------------
>
> In morse_code_decoding.rb
> <https://github.com/chuaindia/Morse_Code_in_Ruby/pull/1#discussion_r1054195757>
> :
>
> > +  ***@***.***} "
>
> +end
>
> +
>
> +def decode(morse_code)
>
> +  @final_word = ''
>
> +  @morse_code_arr = morse_code.split('   ')
>
> +  @morse_code_arr.map do |morse_word|
>
> +    @final_word += decode_word(morse_word)
>
> +  end
>
> +
>
> +  puts ***@***.***_word} "
>
>
>
>    -
>
>    Your methods return unnecessary trailing spaces and your decode method
>    should return result, not print it. Kindly fix that.
>    -
>
>       You can remove the spacing on line 22 to this: *"#{word}"*
>       -
>
>       You can refactor the decode method to this to account for the
>       spacing you are looking for:
>
>       def decode(morse_code)
>
>         @final_word = ''
>
>         @morse_code_arr = morse_code.split('   ')
>
>         @morse_code_arr.map do |morse_word|
>
>           @final_word += decode_word(morse_word) + " "
>
>         end
>
>
>
>         ***@***.***_word}".strip
>
>       end
>
>
>       [image: ygf]
>    <https://user-images.githubusercontent.com/36542986/208876734-40c94ee5-2172-4bfd-a058-c0be3327c503.jpg>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chuaindia/Morse_Code_in_Ruby/pull/1#pullrequestreview-1225793499>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZPANHOEZ62VN5W3CDHBGNLWOLIMPANCNFSM6AAAAAATFNCEB4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

chuaindia committed:
Update file following the code Reviewr's review

chuaindia committed:
Update file following the code Reviewr's review

chuaindia committed:
Update file for linters

chuaindia committed:
Update file for linters

chuaindia committed:
Update file for linters

cynthiainga commented on morse_code_decoding.rb:
```diff
@@ -0,0 +1,35 @@
+  @final_word = ''
+  @morse_code_arr = morse_code.split('   ')
+  @morse_code_arr.map do |morse_word|
+    @final_word += decode_word(morse_word)
```
- [ ] Please kindly add space between your word, as mentioned by the previous reviewer you just need to add space at this line ` @final_word += decode_word(morse_word) + " "`. Kindly fix that 🙃

cynthiainga commented on morse_code_decoding.rb:
```diff
@@ -0,0 +1,35 @@
+  @final_word.to_s.strip
+end
+
+decode('.-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...')
```
- [ ] Also, please kindly add the `p` before `decode` so then your message should return and print correctly 🙃

cynthiainga reviewed (changes requested):
Hi @chuaindia, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

![](https://user-images.githubusercontent.com/80612925/157397398-00904e26-60ec-46cd-9510-240661c250ef.gif)

 ## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

chuaindia committed:
update the rb file following the Reviewr's review

chuaindia committed:
update the rb file following the Reviewr's review

chuaindia committed:
update the rb file following the Reviewr's review

chuaindia committed:
Add updated file for lintres

samuelmoiwa committed:
Update morse_code_decoding.rb

samuelmoiwa committed:
Update morse_code_decoding.rb

mckent05 commented on morse_code_decoding.rb:
```diff
@@ -0,0 +1,36 @@
+  @final_word.to_s.strip
+end
+
+pdecode('.-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...')
```
- Your code doesn't run correctly, that is because you added the `p` keyword suggested by the previous reviewer wrongly, the `p` keyword is used to print out your output. I suggest adding It like this `p decode(...)`. Please fix this 👍 👍

mckent05 reviewed (changes requested):
Hi @chuaindia, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
![almost there](https://user-images.githubusercontent.com/73607512/207265397-5ef597f5-a50c-49b4-8f07-7548b0520e98.gif)


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

samuelmoiwa committed:
Update morse_code_decoding.rb

mckent05 reviewed (approved):
Hi @chuaindia, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
![approved](https://user-images.githubusercontent.com/73607512/207264357-1b05ce6a-95f2-4d31-9c26-888c6ddfa124.gif)


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Yur code works correctly 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
